### PR TITLE
<Firebase Project for each flavor and flavor configuration>

### PR DESCRIPTION
**Pull request by Odaykhz**

### CODE REVIEW

**Comment by hkarmoush** [comment]
.dart tools folder should no show in the commit, it should be added to gitignore to be ignored by git

File: .flutter-plugins [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This is a generated file; do not edit or check into version control.
```
Comment (hkarmoush) [inline .flutter-plugins:1]: This should be ignored too

File: .flutter-plugins-dependencies [hunk 2]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"firebase_core","path":"/Users/odaykhazaeleh/.pub-cache/hosted/pub.dev/firebase_core-2.15.1/","native_build":true,"dependencies":[]}],"android":[{"name":"firebase_core","path":"/Users/odaykhazaeleh/.pub-cache/hosted/pub.dev/firebase_core-2.15.1/","native_build":true,"dependencies":[]}],"macos":[{"name":"firebase_core","path":"/Users/odaykhazaeleh/.pub-cache/hosted/pub.dev/firebase_core-2.15.1/","native_build":true,"dependencies":[]}],"linux":[],"windows":[{"name":"firebase_core","path":"/Users/odaykhazaeleh/.pub-cache/hosted/pub.dev/firebase_core-2.15.1/","native_build":true,"dependencies":[]}],"web":[{"name":"firebase_core_web","path":"/Users/odaykhazaeleh/.pub-cache/hosted/pub.dev/firebase_core_web-2.7.0/","dependencies":[]}]},"dependencyGraph":[{"name":"firebase_core","dependencies":["firebase_core_web"]},{"name":"firebase_core_web","dependencies":[]}],"date_created":"2023-09-17 19:01:14.086265","version":"3.7.12"}
```
Comment (hkarmoush) [inline .flutter-plugins-dependencies:1]: This should be ignored too

File: build/flutter_assets/AssetManifest.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (hkarmoush) [inline build/flutter_assets/AssetManifest.json:1]: build folder should be ignored too

File: ios/Podfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Uncomment this line to define a global platform for your project
+# platform :ios, '11.0'
```
Comment (hkarmoush) [inline ios/Podfile:2]: Let's enable for iOS 13

File: lib/firebase_options_dev.dart [hunk 5]
```diff
@@ -0,0 +1,75 @@
+// File generated by FlutterFire CLI.
```
Comment (hkarmoush) [inline lib/firebase_options_dev.dart:1]: hmmmm,

File: lib/main_production.dart [hunk 6]
```diff
@@ -1,6 +1,11 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
 import 'package:flutter_starter/app/app.dart';
 import 'package:flutter_starter/bootstrap.dart';
+import 'package:flutter_starter/firebase_options_prod.dart';
 
-void main() {
+void main()async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
```
Comment (hkarmoush) [inline lib/main_production.dart:9]: How are you selecting which Firebase project here?

File: lib/main_staging.dart [hunk 7]
```diff
@@ -1,6 +1,11 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
 import 'package:flutter_starter/app/app.dart';
 import 'package:flutter_starter/bootstrap.dart';
+import 'package:flutter_starter/firebase_options_stg.dart';
 
-void main() {
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
```
Comment (hkarmoush) [inline lib/main_staging.dart:9]: How are you selecting which Firebase project here?

Comment (hkarmoush) [inline lib/main_staging.dart:9]: How are you selecting which Firebase project here?

File: pubspec.lock [hunk 8]
```diff
@@ -129,6 +129,30 @@ packages:
       url: "https://pub.dev"
```
Comment (hkarmoush) [inline pubspec.lock:129]: this should be ignored too

File: windows/flutter/generated_plugin_registrant.cc [hunk 9]
```diff
@@ -6,6 +6,9 @@
```
Comment (hkarmoush) [inline windows/flutter/generated_plugin_registrant.cc:6]: remove windows target

File: lib/main_staging.dart (hunk 7 above)
Comment (Odaykhz) [reply lib/main_staging.dart:9]: each firebase project has a firebase_options file for each dev, stg and prod.

Comment (Odaykhz) [reply lib/main_staging.dart:9]: each firebase project has a firebase_options file for each dev, stg and prod.

File: windows/flutter/generated_plugin_registrant.cc (hunk 9 above)
Comment (hkarmoush) [reply windows/flutter/generated_plugin_registrant.cc:6]: @Odaykhz bump

File: pubspec.lock (hunk 8 above)
Comment (hkarmoush) [reply pubspec.lock:129]: @Odaykhz bump

File: build/flutter_assets/AssetManifest.json (hunk 3 above)
Comment (hkarmoush) [reply build/flutter_assets/AssetManifest.json:1]: @Odaykhz bump

File: .flutter-plugins-dependencies (hunk 2 above)
Comment (hkarmoush) [reply .flutter-plugins-dependencies:1]: @Odaykhz bump

File: .flutter-plugins (hunk 1 above)
Comment (hkarmoush) [reply .flutter-plugins:1]: @Odaykhz bump

**Review by hkarmoush — CHANGES_REQUESTED** [review]
review the remaining comments please, good job though

File: lib/main_staging.dart (hunk 7 above)
Comment (hkarmoush) [reply lib/main_staging.dart:9]: Well, You're not passing it yet correctly?

Comment (hkarmoush) [reply lib/main_staging.dart:9]: Ooooh I see, gg


do they run correctly?

Comment (Odaykhz) [reply lib/main_staging.dart:9]: yeah they are good

**Review by hkarmoush — APPROVED** [review]

**Pull request merged**